packages/p2p-media-loader-core/src/segments-storage.ts

```diff
@@ -150,19 +150,22 @@ export class SegmentsMemoryStorage {
     listener: StorageEventHandlers["onStorageUpdated"],
   ) {
     const localId = StreamUtils.getStreamShortId(stream);
-    this.events.subscribe(`onStorageUpdated-${localId}`, listener);
+    this.eventEmitter.addEventListener(`onStorageUpdated-${localId}`, listener);
   }
 
   unsubscribeFromUpdate(
     stream: Stream,
     listener: StorageEventHandlers["onStorageUpdated"],
   ) {
     const localId = StreamUtils.getStreamShortId(stream);
-    this.events.unsubscribe(`onStorageUpdated-${localId}`, listener);
+    this.eventEmitter.removeEventListener(
+      `onStorageUpdated-${localId}`,
+      listener,
+    );
   }
 
   private dispatchStorageUpdatedEvent(stream: Stream) {
-    this.events.dispatch(
+    this.eventEmitter.dispatchEvent(
       `onStorageUpdated${StreamUtils.getStreamShortId(stream)}`,
       stream,
```

Comment:
Test and fix

Reply:
I fixed mistake and now it looks like it works properly
```
 this.segmentStorage.subscribeOnUpdate(
      this.stream,
      this.broadcastAnnouncement,
    );

    this.segmentStorage.subscribeOnUpdate(this.stream, this.testListener);

    this.trackerClient.start();
  }

  testListener = () => {
    console.log("test");
  };

```
![image](https://github.com/Novage/p2p-media-loader/assets/91938357/c43f182c-71a9-426d-9c6a-10fd7657751d)